sources/src/main/java/com/google/solutions/jitaccess/core/MailAddressRegexFormatter.java

```diff
@@ -0,0 +1,58 @@
+package com.google.solutions.jitaccess.core;
+
+import java.util.regex.Pattern;
+import java.util.stream.IntStream;
+
+import com.google.common.base.Preconditions;
+
+public class MailAddressRegexFormatter implements MailAddressFormatter {
+    private final Options options;
+
+    public MailAddressRegexFormatter(Options options) {
+        this.options = options;
+    }
+
+    public record Options(Pattern internalsMailAddressPattern, String internalsMailAddressTransform,
+            Pattern externalsMailAddressPattern, String externalsMailAddressTransform) {
+        public Options {
+            Preconditions.checkNotNull(internalsMailAddressPattern);
+            Preconditions.checkNotNull(internalsMailAddressTransform);
+        }
+    }
```

Comment:
Correct. Perhaps better to provide a list of patterns/strings.

Reply:
Maybe there could be more than two scenarios, and then it would be more generic, sure. But otherwise I think it's fine like this for now - I could see that the formatter was required, but that the ENV vars had defaults, so they don't break any backwards compatibility either way.